metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesFinder.scala

```diff
@@ -0,0 +1,115 @@
+package scala.meta.internal.metals.testProvider
+
+import scala.meta.internal.implementation.ImplementationProvider
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.DefinitionProvider
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.debug.BuildTargetClasses
+
+import ch.epfl.scala.bsp4j.BuildTarget
+
+trait TestSuitesFinder {
+  def findTestSuites(): Seq[TestSuiteDiscoveryResult]
+}
+
+final class TestSuitesFinderImpl(
+    buildTargets: BuildTargets,
+    buildTargetClasses: BuildTargetClasses,
+    definitionProvider: DefinitionProvider,
+    implementationProvider: ImplementationProvider
+) extends TestSuitesFinder {
+
+  override def findTestSuites(): Seq[TestSuiteDiscoveryResult] = {
+    buildTargets.allBuildTargetIds.toList
+      .flatMap(buildTargets.info)
+      .filterNot(_.getDisplayName.endsWith("build"))
```

Comment:
You should instead check if it's sbt build target.

Normal module can also use `build` in their name.

Reply:
Yeah, I know. I was just waiting for a goo advice how to handle that 😎